workshops/07-TPP-Workshop/README.md

```diff
@@ -0,0 +1,82 @@
+# Transformation Priority Premise
+
+This workshop assumes that you have practiced the discipline of Test Driven Development.
+
+We already know that in TDD there are three stages which are repeated until the code is complete.
+
+1. Write the simplest failing test
+2. Write the simplest code to make it pass
+3. Refactor
+
+We also know that before writing a failing test it is **important to think**.
+
+However, we don't want to have to imagine the whole algorithm as that would be counter to the benefits of TDD.
+
+**Remember:** as the tests get more specific, the code gets more generic.
+
+In order to incrementally refactor the code to become more generic we must avoid _getting stuck_. 
+
+In order to ensure that our code is maintainable we must avoid complexity.
+
+## The stages of TDD
+
+We already have a name for first stage of TDD, it's called Testing. We also already have a name for the last stage of TDD, it's called Refactoring.
+
+What we haven't named is the middle stage, "Make it pass".
+
+**What are we doing in that step?** We could say that we are _Transforming code_. 
+
+Leaving us with a nice symmetrical list of names for each stage:
+
+1. Testing
+2. Transforming
+3. Refactoring
+
+## A list of transformation
+
+When we write a failing test we force ourselves to Transform our code to make it pass.
+
+Here is a non-exhaustive list of transformations:
```

Comment:
I go back and forth on whether this big list is good to include.

Do you think most of these terms will be familiar to people taking the workshop? 

Does it help with making the point here to have a few, expanded examples rather than a long list of relatively unexplained ones?